### PR TITLE
Deploy the docs!

### PR DESCRIPTION
**Pull request by lheagy**
- update .travis.yml and app.yaml so that docs with examples deploy
image

- add an `edit_on_github` link to every page to replace the "page source" as the ability to edit is more useful.
image

### CODE REVIEW

**Comment by lheagy** [comment]
This addresses #554 and #560

File: docs/_ext/edit_on_github.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+sidebar.
+
+Loosely based on https://github.com/astropy/astropy/pull/347
+"""
+
+import os
+import warnings
+
+
+def get_github_url(app, view, path):
```
Comment (rowanc1) [inline docs/_ext/edit_on_github.py]: this is cool!

File: docs/app.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-# application: simpegdocs
-# version: 1
+# application: simpeg-dev-docs
```
Comment (rowanc1) [inline docs/app.yaml]: is there a reason for this?

Comment (lheagy) [inline docs/app.yaml]: ah, no I was testing deployment locally and wanted to do that on dev-docs. Would you prefer I update this back to simpegdoc before merging?

**Pull request merged**